pypeit/masterframe.py

```diff
@@ -1,240 +1,221 @@
-        hdulist[0].header["FRAMETYP"] = (self.frametype, 'PyepIt: Master calibration frame type')
-        if keywds is not None:
-            for key in keywds.keys():
-                hdulist[0].header[key] = keywds[key]
-        # Write the file to disk
-        if os.path.exists(_outfile):
-            msgs.warn("Overwriting file:" + msgs.newline() + _outfile)
-
-        hdulist.writeto(_outfile, overwrite=True)
-
-        # Finish
-        msgs.info("Master {0:s} frame saved successfully:".format(self.frametype) + msgs.newline() + _outfile)
-        return
-
-
-def master_name(ftype, master_key, mdir):
-    """ Default filenames for MasterFrames
-
-    Args:
-        ftype (str):
-          Frame type
-        master_key (str):
-          Setup name, e.b. A_1_01
-        mdir (str):
-          Master directory
-
-    Returns:
-        str: masterframe filename
-    """
-    name_dict = dict(bias='{:s}/MasterBias_{:s}.fits'.format(mdir, master_key),
-                     badpix='{:s}/MasterBadPix_{:s}.fits'.format(mdir, master_key),
-                     trace='{:s}/MasterTrace_{:s}.fits'.format(mdir, master_key),
-                     pinhole='{:s}/MasterPinhole_{:s}.fits'.format(mdir, master_key),
-                     pixelflat='{:s}/MasterPixelFlat_{:s}.fits'.format(mdir, master_key),
-                     illumflat='{:s}/MasterIllumFlat_{:s}.fits'.format(mdir, master_key),
-                     arc='{:s}/MasterArc_{:s}.fits'.format(mdir, master_key),
-                     wave='{:s}/MasterWave_{:s}.fits'.format(mdir, master_key),
-                     wv_calib='{:s}/MasterWaveCalib_{:s}.json'.format(mdir, master_key),
-                     tilts='{:s}/MasterTilts_{:s}.fits'.format(mdir, master_key),
-                     # sensfunc='{:s}/MasterSensFunc_{:s}_{:s}.yaml'.format(mdir, master_key[0], master_key[-2:]),
-                     sensfunc='{:s}/MasterSensFunc_{:s}_{:s}.fits'.format(mdir, master_key[0], master_key[-2:]),
-                     )
-    return name_dict[ftype]
-
-
-def set_master_dir(redux_path, spectrograph, par):
-    """
-    Set the master directory auto-magically
-
-    Args:
-        redux_path: str or None
-        spectrograph: Spectrograph or None
-        par: ParSet or None
-
-    Returns:
-        master_dir : str
-          Path of the MasterFrame directory
-
-    """
-    # Parameters
-    if par is None:
-        tmppar = pypeitpar.CalibrationsPar()
-    else:
-        if 'caldir' not in par.keys():
-            tmppar = pypeitpar.CalibrationsPar()
-        else:
-            tmppar = par
-    # Redux path
-    if redux_path is None:
-        redux_path = os.getcwd()
-    master_dir = os.path.join(redux_path, tmppar['caldir'])
-    # Spectrograph
-    if spectrograph is not None:
-        master_dir += '_'+spectrograph.spectrograph
-    # Return
-    return master_dir
-
+        raw_files = {}
+        prefix = 'F'
+        for k, v in hdr.items():
```

Comment:
Having a hard time parsing this.  Please add a comment on what
is being done.

Reply:
Done.  Let me know if it's still unclear.